### PR TITLE
feat: jupyter lab 3.0 federated bundle support

### PR DESCRIPTION
**Pull request by maartenbreddels**
pip install, and it works 🎉

### CODE REVIEW

**Comment by maartenbreddels** [comment]
Uses https://github.com/bqplot/bqplot/pull/1222 as a template

File: js/package.json [hunk 1]
```diff
@@ -53,6 +54,16 @@
...
     "extension": "src/jupyterlab-plugin",
+    "outputDir": "../share/jupyter/labextensions/jupyter-threejs",
+    "sharedPackages": {
+      "@jupyter-widgets/base": {
+        "bundled": false,
+        "singleton": true
+      },
+      "threejs": {
+        "bundled": true
+      }
```
Comment (jasongrout) [inline js/package.json:65]:
```suggestion
      }
```
The default is to bundle, so this does not have to be specified.

File: js/src/jupyterlab-plugin.js [hunk 2]
```diff
@@ -19,7 +19,7 @@ module.exports = {
                             console.error(err);
                             reject(err);
                         },
-                        'jupyter-threejs'
+                        'jupyter-threejs-chunk'
```
Comment (jasongrout) [inline js/src/jupyterlab-plugin.js:22]: This change is surprising to me. Why do  you need `-chunk`?

Comment (maartenbreddels) [reply js/src/jupyterlab-plugin.js:22]: the chunk name was not allowed to be the same name as package/module/forgotwhatitwasname, as long as it was not `jupyter-threejs` it was fine.

**Comment by agoose77** [comment]
Is this nearly ready to go? Is there anything else that I can help with? I made a small PR on this branch to use jupyter-packaging: https://github.com/maartenbreddels/pythreejs/pull/1/files

File: setup.py [hunk 3]
```diff
@@ -39,6 +41,12 @@
         ('share/jupyter/lab/extensions',
          'js/lab-dist',
          'jupyter-threejs-*.tgz'),
+        ('share/jupyter/labextensions/jupyter-threejs/',
```
Comment (jasongrout) [inline setup.py:44]: It's easier to use `**`, like https://github.com/jupyterlab/extension-cookiecutter-ts/blob/master/%7B%7Bcookiecutter.python_name%7D%7D/setup.py#L41-L42

**Comment by agoose77** [comment]
@maartenbreddels Are you happy with these changes to your PR? https://github.com/maartenbreddels/pythreejs/pull/2

**Comment by bollwyvl** [comment]
Can confirm, this PR looks good to go (works locally). I'd wager bumping the nodejs to something not EOL (or about to be) would clean up the CI fail... or just chuck it and head to github actions (which I'd be happy to PR).

**Comment by maartenbreddels** [comment]
Thanks, I think we only need to agree on https://github.com/maartenbreddels/pythreejs/pull/2

File: js/src/jupyterlab-plugin.js (hunk 2 above)
Comment (bollwyvl) [reply js/src/jupyterlab-plugin.js:22]: can confirm, have encountered this elsewhere

File: setupbase.py [hunk 4]
```diff
@@ -1,735 +0,0 @@
-#!/usr/bin/env python
```
Comment (bollwyvl) [inline setupbase.py:1]: :tada:

File: js/package.json [hunk 5]
```diff
@@ -53,6 +54,13 @@
   },
   "jupyterlab": {
     "extension": "src/jupyterlab-plugin",
+    "outputDir": "../share/jupyter/labextensions/jupyter-threejs",
+    "sharedPackages": {
+      "@jupyter-widgets/base": {
+        "bundled": false,
+        "singleton": true
+      }
+    },
```
Comment (bollwyvl) [inline js/package.json:63]: It might be nice to hoist threejs as a shared package (non-singleton, ha!)

File: setupbase.py [hunk 6]
```diff
@@ -1,735 +0,0 @@
...
-
-
-
-# pythreejs specific functions to handle autogen:
-
-# Call this after running wrapper command, but before running wrapped command:
-
-def update_packages(command):
-    """update build_py options to get packages changes"""
-    command.distribution.packages = find_packages()
```
Comment (vidartf) [inline setupbase.py:735]: This bit is critical!

Comment (agoose77) [reply setupbase.py:735]: This one I actually missed, thanks for the catch. Will fix

**Comment by agoose77** [comment]
Just a note: this is probably waiting on https://github.com/maartenbreddels/pythreejs/pull/2 

Apologies for some of the commit noise.

**Pull request closed without merging**